File: 00xx/002x/0026-RemoveDuplicatesFromSortedArray/Solution.go
/*  Solution: Slicing
    Time Complexity:    O(n)
    Space Complexity:   O(1)
 */

func removeDuplicates(nums []int) int {
  i := 1
  for i < len(nums) {
    if nums[i] == nums[i-1] {
      nums = append(nums[:i], nums[i+1:]...)
    } else {
      i++
    }
  }
  return i
}

File: 00xx/002x/0026-RemoveDuplicatesFromSortedArray/Solution.py
# Solution: Check previous index
# Time Complexity:  O(n)
# Space Complexity: O(1)

class Solution:
  def removeDuplicates(self, nums: List[int]) -> int:
    i = 1
    while i < len(nums):
      if nums[i-1] == nums[i]:
        del nums[i]
      else:
        i += 1
    return len(nums)
